core/Test/Tasty/Ingredients/ConsoleReporter.hs

```diff
@@ -175,6 +201,12 @@ buildTestOutput opts tree =
           }
           opts tree
 
+-- | Make sure the progress text does not contain any newlines or line feeds,
+-- lest our ANSI magic breaks. Since the progress text is expected to be short,
```

Comment:
Are `'\t'` fine? I imagine they can also break ANSI magic.

Reply:
I don't know. I don't mind removing them as well, it's rather unexpected to push `\t`'s to progress info.

Reply:
I think it's better to replace every `Data.Char.isSpace` character with `' '`.